test/dialing/direct.node.js

```diff
@@ -235,6 +241,28 @@ describe('Dialing (direct, TCP)', () => {
 
     after(() => remoteLibp2p.stop())
 
+    it('should fail if no peer id is provided', async () => {
+      libp2p = new Libp2p({
+        peerInfo,
+        modules: {
+          transport: [Transport],
+          streamMuxer: [Muxer],
+          connEncryption: [Crypto]
+        }
+      })
+
+      sinon.spy(libp2p.dialer, 'connectToPeer')
+
+      try {
+        await libp2p.dial(remoteLibp2p.transportManager.getAddrs()[0])
+      } catch (err) {
+        expect(err).to.have.property('code', ErrorCodes.ERR_INVALID_PEER)
```

Comment:
Should we divide this error in 2 different ones?

Throwing to the user an invalid peer error, when the peer id was not provided might lead to confusion. I would separate between an invalid peerID and a not provided peerID error. What do you think?

Reply:
The message that gets returned with the error is `The multiaddr did not contain a valid peer id`. It doesn't necessarily mean that no peer id was provided, it could also mean that the peer id provided isn't supported. Are you suggesting that if no peer id is provided we return one error, and if a peer id is provided but it's a bad peer id we return a different error? If so, I think it might be excessive.

Reply:
> Are you suggesting that if no peer id is provided we return one error, and if a peer id is provided but it's a bad peer id we return a different error? If so, I think it might be excessive.

Yes, that was my suggestion. I am just concerned that a new user (who is also does not have knowledge on multiaddrs) might not understand immediately that the error originated from a multiaddr with no `peer-id`. However, from the error msg the user can understand that, so yeah I agree that this might be excessive.

Other suggestion. Should we add this requirements In the api docs? We just say that `libp2p.dial` can receive the following types of parameter: `PeerInfo|PeerId|Multiaddr|string`. Being a multiaddr valid without a peer id, but then throwing invalid peer id when no peer id is provided seems strange. Meanwhile, I will approve this PR, as this is a small detail

Reply:
I updated the description in the api for dial and dialProtocol to read: 
"The peer to dial. If a [`Multiaddr`][multiaddr] or its string is provided, it **must** include the peer id"